contracts/src/TokenSpoke/ERC20TokenSpoke.sol

```diff
@@ -159,4 +160,27 @@ contract ERC20TokenSpoke is IERC20TokenBridge, ERC20, TokenSpoke {
             _transfer(address(this), message.fallbackRecipient, remainingAllowance);
         }
     }
+
+    /**
+     * @notice See {TokenSpoke-_handleFees}
+     *
+     * If the {feeTokenAddress} is this contract, use the internal ERC20 calls
+     * to transfer the tokens directly. Otherwise, use the {SafeERC20TransferFrom} library
+     * to transfer the tokens.
+     */
+    function _handleFees(
+        address feeTokenAddress,
+        uint256 feeAmount
+    ) internal override returns (uint256) {
+        if (feeAmount == 0) {
+            return 0;
+        }
+        // If the {feeTokenAddress} is this contract, then just deposit the tokens directly.
+        if (feeTokenAddress == address(this)) {
+            _spendAllowance(_msgSender(), address(this), feeAmount);
+            _transfer(_msgSender(), address(this), feeAmount);
+            return feeAmount;
```

Comment:
Doesn't matter too much either way, but we could just call `_deposit` here I think still.

Reply:
Decided against it because wanted to avoid developers assuming they can safely call `_deposit` there all the time and making a similar mistake for `NativeTokenSpoke` implementations.